testsuite/features/step_definitions/retail_steps.rb

```diff
@@ -42,15 +42,129 @@
+    #                  otherwise the name server we get from DHCP is lost at the end of the list
+    #                  (the name servers list in resolv.conf is limited to 3 entries)
+    cmd = 'nmcli connection modify "Wired connection 1" ipv4.dns-priority 20 && ' \
+          'nmcli device modify eth0 ipv4.dns-priority 20 && ' \
+          'nmcli connection modify "Wired connection 2" ipv4.dns-priority 10 && ' \
+          'nmcli device modify eth1 ipv4.dns-priority 10'
+  else
+    # Wicked: is there a way to give eth1 precedence?
+    #         we use a static setting for the name server instead
+    static_dns = net_prefix + PRIVATE_ADDRESSES['dhcp_dns']
+    cmd = 'echo -e "BOOTPROTO=dhcp\nSTARTMODE=auto\n" > /etc/sysconfig/network/ifcfg-eth1 && ' \
+          'ifup eth1 && ' \
+          "sed -i 's/^NETCONFIG_DNS_STATIC_SERVERS=\".*\"/NETCONFIG_DNS_STATIC_SERVERS=\"#{static_dns}\"/' /etc/sysconfig/network/config && " \
+          'netconfig update -f'
+  end
+  node.run(cmd)
+end
+
+Then(/^the "([^"]*)" host should be present on private network$/) do |host|
+  node = get_target('proxy')
+  output, return_code = node.run("ping -n -c 1 -I #{node.private_interface} #{net_prefix}#{PRIVATE_ADDRESSES[host]}")
+  raise SystemCallError, "Terminal #{host} does not answer on eth1: #{output}" unless return_code.zero?
+end
+
+Then(/^name resolution should work on private network$/) do
+  node = get_target('proxy')
+  # direct name resolution
+  %w[proxy.example.org dns.google.com].each do |dest|
+    output, return_code = node.run("host #{dest}", check_errors: false)
+    raise SystemCallError, "Direct name resolution of #{dest} on proxy doesn't work: #{output}" unless return_code.zero?
+
+    log output.to_s
   end
-  get_target('server').run("ping -c 1 #{node.public_ip}")
+  # reverse name resolution
+  [node.private_ip, '8.8.8.8'].each do |dest|
+    output, return_code = node.run("host #{dest}", check_errors: false)
+    raise SystemCallError, "Reverse name resolution of #{dest} on proxy doesn't work: #{output}" unless return_code.zero?
+
+    log output.to_s
+  end
+end
+
+When(/^I restart the network on the PXE boot minion$/) do
+  # We have no IPv4 address on that machine yet,
+  # so the only way to contact it is via IPv6 link-local.
+  # We convert MAC address to IPv6 link-local address:
+  mac = $pxeboot_mac.tr(':', '')
+  hex = (("#{mac[0..5]}fffe#{mac[6..11]}").to_i(16) ^ 0x0200000000000000).to_s(16)
+  ipv6 = "fe80::#{hex[0..3]}:#{hex[4..7]}:#{hex[8..11]}:#{hex[12..15]}%eth1"
+  file = 'restart-network-pxeboot.exp'
+  source = "#{File.dirname(__FILE__)}/../upload_files/#{file}"
+  dest = "/tmp/#{file}"
+  return_code = file_inject(get_target('proxy'), source, dest)
+  raise ScriptError, 'File injection failed' unless return_code.zero?
+
+  # We have no direct access to the PXE boot minion
+  # so we run the command from the proxy
+  get_target('proxy').run("expect -f /tmp/#{file} #{ipv6}")
```

Comment:
I think in this case we want to run it from the container, yes.

I will double-check that `inject` works with containers too. But I doubt it doesn't, otherwise it would not have worked during my tests.

Reply:
sorry, my answer was wrong

correct answer is: on the proxy, everything is done outside of containers...